### PR TITLE
Default to containerless mode

### PR DESCRIPTION
**Pull request by eemcmullan**
Fixes https://issues.redhat.com/browse/MTA-4217 and https://issues.redhat.com/browse/MTA-4219

### CODE REVIEW

File: cmd/analyze.go [hunk 1]
```diff
@@ -200,12 +201,9 @@ func NewAnalyzeCmd(log logr.Logger) *cobra.Command {
 			defer stop()
 
 			// ***** RUN CONTAINERLESS MODE *****
-			if Settings.RunLocal {
-				if analyzeCmd.listProviders {
```
Comment (aufi) [inline cmd/analyze.go:204]: --list-providers should return instead of starting analysis with container-less/run_local.

File: cmd/analyze.go [hunk 2]
```diff
@@ -378,6 +376,7 @@ func NewAnalyzeCmd(log logr.Logger) *cobra.Command {
 	analyzeCommand.Flags().StringArrayVarP(&analyzeCmd.depFolders, "dependency-folders", "d", []string{}, "directory for dependencies")
 	analyzeCommand.Flags().StringVar(&analyzeCmd.overrideProviderSettings, "override-provider-settings", "", "override the provider settings, the analysis pod will be run on the host network and no providers will be started up")
 	analyzeCommand.Flags().StringArrayVar(&analyzeCmd.provider, "provider", []string{}, "specify which provider(s) to run")
+	analyzeCommand.Flags().BoolVar(&analyzeCmd.runLocal, "run-local", true, "run Java anlysis in containerless mode")
```
Comment (aufi) [inline cmd/analyze.go:379]: Nit, typo in command description (anlysis).

**Review by aufi — COMMENTED** [review]
Thanks for the PR! Left few comments, CI probably needs some care, otherwise looks good.

General note on the default containerless, since there is also default analyze mode "full (to analyze source+dependencies), analysis results might be different than before since there is persisted Maven .m2 cache with packages. So, even without adding settings.xml file to the analyze command, some packages might be already present in the local maven repository cache and some rules can match on those.

**Comment by eemcmullan** [comment]
@aufi I agree. I think that's a good thing to be aware of and look out for. May need to discuss some solutions next week, because I'm not 100% sure on a good way to solve that.

**Review by aufi — APPROVED** [review]
LGTM

**Pull request merged**